splitio/sync/split.py

```diff
@@ -318,3 +337,54 @@ def synchronize_splits(self, till=None):  # pylint:disable=unused-argument
 
         for split in to_delete:
             self._split_storage.remove(split)
+
+        return None
+
+    def _synchronize_json(self):
+        """Update splits in storage for json mode."""
+        if not self._filename.lower().endswith(('.json')):
+            raise ValueError("json File provided %s does not have .json extension." % self._filename)
+
+        fetched = self._read_splits_from_json_file(self._filename)
+        segment_list = set()
+        if self._current_json_sha == "-1" or self._get_sha(json.dumps(fetched)) != self._current_json_sha:
```

Comment:
same as segments, changenumber is timestamp, sha should be something that you track inside this instance

Reply:
added check for split changeNumber in addition to sha